x/concentrated-liquidity/position.go

```diff
@@ -380,6 +398,23 @@ func (k Keeper) validatePositionsAndGetTotalLiquidity(ctx sdk.Context, owner sdk
 			return 0, 0, 0, sdk.Dec{}, types.PositionOwnerMismatchError{PositionOwner: position.Address, Sender: owner.String()}
 		}
 
+		// Check that all the positions have no underlying lock that has not yet matured.
+		positionHasUnderlyingLock := k.positionHasUnderlyingLockInState(ctx, positionId)
+		if positionHasUnderlyingLock {
+			underlyingLockId, err := k.GetPositionIdToLock(ctx, positionId)
+			if err != nil {
+				return 0, 0, 0, sdk.Dec{}, err
+			}
+
+			lockIsMature, err := k.isLockMature(ctx, underlyingLockId)
+			if err != nil {
+				return 0, 0, 0, sdk.Dec{}, err
+			}
+			if !lockIsMature {
+				return 0, 0, 0, sdk.Dec{}, types.LockNotMatureError{LockId: underlyingLockId}
+			}
```

Comment:
Not seeing test cases added for this logic, please add them

Reply:
The test case for validatePositionsAndGetTotalLiquidity is included in the fungify charge position test here https://github.com/osmosis-labs/osmosis/blob/97caf31bafde3d635cb7e4001d6222d441619c5d/x/concentrated-liquidity/position_test.go#L570